src/cpp/r/RCntxt.cpp

```diff
@@ -111,12 +111,24 @@ SEXP RCntxt::originalFunctionCall() const
 
 Error RCntxt::fileName(std::string* pFileName) const
 {
-   if (srcref() && TYPEOF(srcref()) != NILSXP)
+   // skip over bytecode srcrefs
+   RCntxt context = *this;
+   SEXP ref = context.srcref();
+   while (ref && isByteCodeSrcRef(ref))
```

Comment:
Two questions on this one:

a) How do we know that the `srcref` we skip to refers to the same file as the one on which `fileName` was invoked (since we skip an arbitrary number)?

b) Besides the filename, is there any other information for which we might want this bytecode-skipping behavior? I'm wondering in particular if methods such as e.g. `sourceRefs` need to seek to the "real" srcref in this manner.